### ISSUE TITLE
Training error percentage worsened with higher iterations.

### ISSUE THREAD

**Issue by sbr2567** [BrainJS/brain.js#544]
So I ran some training data in brain.recurrent.LSTM to analysis strings. The log was set to true and 20000 iterations were used. The first 1,000 I was averaging around a 0.035 error percentage. however, 13,000 iterations in, it's now averaging 0.045 with the max hitting INFINITY... Not just that, the training error percentage never went below 0.03. Once it got to 0.03 it kept reverting back up to 0.04. Is there something i'm doing wrong here because the docs say the training error percentage is suppose to improve over time? It's gotten worse.

```
const brain = require('brain.js')
const data = require('./training-data')

const network = new brain.recurrent.LSTM();

const trainingData = data.map(item => ({
    input: item.brand,
    output: item.type
}));

network.train(trainingData, {
    iterations: 20000,
    log: true,
    logPeriod: 5
    
});

const output = network.run("NMD R1 Torch Shoes");
```
At the start
```
iterations: 430, training error: 0.03384511155627626
iterations: 435, training error: 0.03247143961425694
iterations: 440, training error: 0.03274632633124924
iterations: 445, training error: 0.03201798340133459
iterations: 450, training error: 0.03154434072065176
iterations: 455, training error: 0.03152404580442265
iterations: 460, training error: 0.03162264807225351
iterations: 465, training error: 0.03292514434534933
iterations: 470, training error: 0.031105266578095262
iterations: 475, training error: 0.03054067946988782
iterations: 480, training error: 0.0400283211722825
iterations: 485, training error: 0.03816742727976204
iterations: 490, training error: 0.034856846664717875
iterations: 495, training error: 0.0346398714014905
iterations: 500, training error: 0.033456940818593656
iterations: 505, training error: 0.035355135606936434
iterations: 510, training error: 0.04220573422969472
iterations: 515, training error: 0.03635907966351159
iterations: 520, training error: 0.03702175625611862
```
8 hours later.
```
iterations: 13650, training error: 0.043724036322546386
iterations: 13655, training error: 0.04250799821339848
iterations: 13660, training error: 0.04443141732586302
iterations: 13665, training error: 0.04073663554771578
iterations: 13670, training error: 0.040702198922368434
iterations: 13675, training error: 0.04082752818860178
iterations: 13680, training error: 0.039619097415735986
iterations: 13685, training error: Infinity
iterations: 13690, training error: 0.121674811019806
iterations: 13695, training error: 0.10521295302067404
iterations: 13700, training error: 0.08677389942398557
iterations: 13705, training error: 0.0812407006256091
iterations: 13710, training error: 0.06672014295963603
iterations: 13715, training error: 0.06051236516507974
iterations: 13720, training error: 0.05703390458619305
iterations: 13725, training error: 0.053103925922298165
iterations: 13730, training error: 0.05055664988865909
iterations: 13735, training error: 0.0643592753255508
iterations: 13740, training error: 0.06477116810257699
iterations: 13745, training error: 0.05529440501915551
iterations: 13750, training error: 0.05243921742504593
iterations: 13755, training error: 0.049163836806871884
```
UPDATE:
Now the error percentage is stuck at Infinity.
```
iterations: 13995, training error: 0.03830093105323607
iterations: 14000, training error: 0.04068065445059517
iterations: 14005, training error: 0.04871714650130028
iterations: 14010, training error: Infinity
iterations: 14015, training error: Infinity
iterations: 14020, training error: Infinity
iterations: 14025, training error: Infinity
iterations: 14030, training error: Infinity
iterations: 14035, training error: Infinity
iterations: 14040, training error: Infinity
iterations: 14045, training error: Infinity
iterations: 14050, training error: Infinity
iterations: 14055, training error: Infinity
iterations: 14060, training error: Infinity
iterations: 14065, training error: Infinity
iterations: 14070, training error: Infinity
```
How can higher iterations completely result in unusable learning.

**Comment by robertleeplummerjr** [comment]
It sounds like the network size and learning rate need to be adjusted.

**Comment by sbr2567** [comment]
Okay, so would you recommend a slower learning rate?

**Comment by robertleeplummerjr** [comment]
Probably smaller, but also bigger net to retain the added information.  Sometimes you can go deeper for things like this, but ultimately it will require experimentation.

**Comment by sbr2567** [comment]
Okay great thanks, i'll see how I go.
